app/views/foreman/job_templates/install_packages_by_search_query_-_katello_ansible_default.erb

```diff
@@ -0,0 +1,26 @@
+<%#
+kind: job_template
+name: Install packages by search query - Katello Ansible Default
+job_category: Katello
+description_format: 'Install package(s) %{Package search query}'
+feature: katello_package_install_by_search
+provider_type: Ansible
+template_inputs:
+- name: Package search query
+  description: Filter criteria for packages to be installed. IMPORTANT- If left blank, the job will attempt to install all possible packages.
+  input_type: user
+  required: false
+%>
+<% package_names = @host.package_names_for_job_template(
+  action: 'install',
+  search: input('Package search query')
+) -%>
+---
+- hosts: all
+  tasks:
+    - package:
+        name: 
+<% package_names.each do |package_name| -%>
+          - <%= package_name %>
+<% end -%>
```

Comment:
```suggestion
<% if package_names.empty? -%>
        name: []
<% else -%>
        name:
<% package_names.each do |package_name| -%>
          - <%= package_name %>
<% end -%>
<% end -%>
```

Oh one thing about short form/long form for list: if list is empty (because we use the REX tab) [you need to use the short form](https://stackoverflow.com/a/5110341/7227940) otherwise the yaml won't be valid and we will get an syntax error.

So either print only short for the empty case or render_error that forbid it.